node/src/proxy_server/http_protocol_pack.rs

```diff
@@ -91,10 +94,10 @@ impl HttpProtocolPack {
         )
     }
 
-    fn port_from_string(port_str: String) -> Option<u16> {
+    fn port_from_string(port_str: String) -> Result<u16, String> {
         match port_str.parse::<u16>() {
-            Err(_) => None,
-            Ok(port) => Some(port),
+            Err(_) => Err(format!("Port '{}' is not a number", port_str)),
+            Ok(port) => Ok(port),
```

Comment:
If we can't parse a port number, that's not just "no port was specified;" that's an error that needs to be reported and logged somewhere.